Make mantidpython work when python is python3

rosswhitfield opened the pull request:
mantidpython doesn't run when python points to python3, this forces it to use python2

Now that systemtests use mantidpython this has become more important.

*Testing*:
Only code review needed.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

stuartcampbell commented:
I think that this looks sensible - should be ok :shipit:

martyngigg commented on buildconfig/CMake/Packaging/mantidpython.in:
```diff
@@ -7,7 +7,7 @@
 SCRIPTFILE=$(readlink -f "$0")
 INSTALLDIR=$(echo $SCRIPTFILE | sed -r -e 's|^(.*)/(.*)$|\1|g') #.* is greedy and eats up until the final slash
-PROG=$(command -v ipython)
+PROG=$(command -v ipython2)
```
On Ubuntu 14.04 I don't have an `ipython2` command. I have only `ipython3` & `ipython`.

peterfpeterson replied:
You can `if [ $(command -v ipython2) ]; then` around this and fall through to plain `ipython` in the else clause.

rosswhitfield commented:
Dammit. I checked RHEL7 which has `ipython2`, I see ubuntu 14.04 has `ipython2.7` but that doesn't help. Maybe I will need to try `ipython2` first and if it can't find it use `ipython`. Unless there is a smarter way to selected ipython version.

martyngigg commented:
@rosswhitfield I don't know of a better way.

rosswhitfield commented:
Updated. Although I feel like there is a cleaner way to do this in bash.

I also changed the python executable to use `@PYTHON_EXECUTABLE@` from cmake which should always be correct, just in case `python2` doesn't exist somewhere.
